R/imputation.R

```diff
@@ -0,0 +1,326 @@
+##'
+##' @aliases imputeMethods impute_neighbour_average impute_knn impute_mle impute_bpca impute_mixed impute_min impute_zero impute_matrix
+##'
+##' @importFrom Rcpp sourceCpp
+##' 
+##' @useDynLib MsCoreUtils
+##'
+##' @examples
+##'
+##' ## test data
+##' set.seed(42)
+##' m <- matrix(rlnorm(60), 10)
+##' dimnames(m) <- list(letters[1:10], LETTERS[1:6])
+##' m[sample(60, 10)] <- NA
+##' 
+##' ## available methods
+##' imputeMethods()
+##'
+##' impute_matrix(m, method = "zero")
+##'
+##' impute_matrix(m, method = "min")
+##'
+##' impute_matrix(m, method = "knn")
+##'
+##'
+##' ## all but third and fourth features' missing values 
+##' ## are the result of random missing values
+##' randna <- rep(TRUE, 10)
+##' randna[c(3, 9)] <- FALSE
+##' 
+##' impute_matrix(m, method = "mixed",
+##'               randna = randna,
+##'               mar = "knn",
+##'               mnar = "min")
+##' 
+##' @param x A matrix with missing values to be imputed.
+##' 
+##' @param method `character(1)` defining the imputation method. See
+##'     `imputeMethods()` for available ones.
+##' 
+##' @param ... Additional parameters passed to the inner imputation
+##'     function.
+##'
+##' @export
+impute_matrix <- function(x,
+                          method,
+                          ...) {
+    if (!anyNA(x)) return(x)
+    if (missing(method))
+        stop("Please specify an imputation method. ",
+             "See '?impute' for details.")
+    method <- match.arg(method,
+                        choices = imputeMethods(),
+                        several.ok = FALSE)
+    res <- x
+    if (method %in% c("CRILC", "MinDet", "MinProb"))
+        requireNamespace("imputeLCMD")
+    if (method == "knn") {
+        res <- impute_knn(x, ...)
+    } else if (method == "nbavg") {
+        res <- impute_neighbour_average(x, ...)
+    } else if (method == "MLE") {
+        res <- impute_mle(x, ...)
+    } else if (method == "bpca"){
+        res <- impute_bpca(x, ...)
+    } else if (method == "QRILC") {
+        res <- imputeLCMD::impute.QRILC(x, ...)[[1]]
+    } else if (method == "MinDet") {
+        res <- imputeLCMD::impute.MinDet(x, ...)
+    } else if (method == "MinProb") {
+        res <- imputeLCMD::impute.MinProb(x, ...)
+    } else if (method == "min") {
+        res <- impute_min(x)
+    } else if (method == "mixed") {
+        res <- impute_mixed(x, ...)
+    } else if (method == "zero") {
+        res <- impute_zero(x)
+    }
+    ## else method == "none" -- do nothing
+    res
+}
+
+
+##' @export
+##' @rdname imputation
+imputeMethods <- function()
+    c("bpca","knn", "QRILC", "MLE",
+      "MinDet", "MinProb", "min", "zero",
+      "mixed", "nbavg", "none")
+
+##' @export
+##' @rdname imputation
+##' @param k `numeric(1)` providing the imputation value used for the
+##'     first and last samples if they contain an `NA`. The default is
+##'     to use the smallest value in the data.
+impute_neighbour_average <- function(x, k = NULL) {    
```

Comment:
oh yes! it's snake_case time! wondering what @sgibb will say to this :) I personally like it this way, especially since function names like `imputeKnn` would look strange...

Reply:
:-)